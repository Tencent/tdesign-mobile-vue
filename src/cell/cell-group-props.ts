/* eslint-disable */

/**
 * 该文件为脚本自动生成文件，请勿随意修改。如需修改请联系 PMC
 * */

import { TdCellGroupProps } from '../cell/type';
import { PropType } from 'vue';

export default {
  /** 是否显示组边框 */
  bordered: Boolean,
  /** 单元格组风格 */
  theme: {
    type: String as PropType<TdCellGroupProps['theme']>,
    default: 'default' as TdCellGroupProps['theme'],
    validator(val: TdCellGroupProps['theme']): boolean {
      if (!val) return true;
      return ['default', 'card'].includes(val);
    },
  },
  /** 单元格组标题 */
  title: {
    type: String,
    default: '',
  },
};
