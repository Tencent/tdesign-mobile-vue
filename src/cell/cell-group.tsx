import { computed, defineComponent } from 'vue';
import props from './cell-group-props';
import config from '../config';
import { usePrefixClass } from '../hooks/useClass';
import { useTNodeJSX } from '../hooks/tnode';

const { prefix } = config;

export default defineComponent({
  name: `${prefix}-cell-group`,
  props,
  setup(props) {
    const renderTNodeJSX = useTNodeJSX();
    const cellGroupClass = usePrefixClass('cell-group');

    const contentClasses = computed(() => [
      cellGroupClass.value,
      `${cellGroupClass.value}--${props.theme}`,
      { [`${cellGroupClass.value}--bordered`]: props.bordered },
    ]);

    return () => {
      const title = renderTNodeJSX('title');
      return (
        <div>
          {title ? <div class={`${cellGroupClass.value}__title`}>{title}</div> : null}
          <div class={contentClasses.value}>{renderTNodeJSX('default')}</div>
        </div>
      );
    };
  },
});
