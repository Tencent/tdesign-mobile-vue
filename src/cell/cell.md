:: BASE_DOC ::

## API
### Cell Props

名称 | 类型 | 默认值 | 说明 | 必传
-- | -- | -- | -- | --
align | String | middle | 内容的对齐方式，默认居中对齐。可选项：top/middle/bottom | N
arrow | Boolean | false | 是否显示右侧箭头 | N
bordered | Boolean | true | 是否显示下边框 | N
description | String / Slot / Function | - | 下方内容描述。TS 类型：`string \| TNode`。[通用类型定义](https://github.com/Tencent/tdesign-mobile-vue/blob/develop/src/common.ts) | N
hover | Boolean | - | 是否开启点击反馈 | N
image | String / Slot / Function | - | 主图。TS 类型：`string \| TNode`。[通用类型定义](https://github.com/Tencent/tdesign-mobile-vue/blob/develop/src/common.ts) | N
leftIcon | Slot / Function | - | 左侧图标，出现在单元格标题的左侧。TS 类型：`TNode`。[通用类型定义](https://github.com/Tencent/tdesign-mobile-vue/blob/develop/src/common.ts) | N
note | String / Slot / Function | - | 和标题同行的说明文字。TS 类型：`string \| TNode`。[通用类型定义](https://github.com/Tencent/tdesign-mobile-vue/blob/develop/src/common.ts) | N
required | Boolean | false | 是否显示表单必填星号 | N
rightIcon | Slot / Function | - | 最右侧图标。TS 类型：`TNode`。[通用类型定义](https://github.com/Tencent/tdesign-mobile-vue/blob/develop/src/common.ts) | N
title | String / Slot / Function | - | 标题。TS 类型：`string \| TNode`。[通用类型定义](https://github.com/Tencent/tdesign-mobile-vue/blob/develop/src/common.ts) | N
onClick | Function |  | TS 类型：`(context: { e: MouseEvent }) => void`<br/>右侧内容。TS 类型：`string \| TNode`。[通用类型定义](https://github.com/Tencent/tdesign-mobile-vue/blob/develop/src/common.ts) | N

### Cell Events

名称 | 参数 | 描述
-- | -- | --
click | `(context: { e: MouseEvent })` | 右侧内容。TS 类型：`string \| TNode`。[通用类型定义](https://github.com/Tencent/tdesign-mobile-vue/blob/develop/src/common.ts)

### CellGroup Props

名称 | 类型 | 默认值 | 说明 | 必传
-- | -- | -- | -- | --
bordered | Boolean | false | 是否显示组边框 | N
theme | String | default | `0.15.0`。单元格组风格。可选项：default/card | N
title | String | - | 单元格组标题 | N


### CSS 变量
组件提供了下列 CSS 变量，可用于自定义样式。
名称 | 默认值 | 描述 
-- | -- | --
--td-cell-bg-color | @bg-color-container | - 
--td-cell-border-color | @component-stroke | - 
--td-cell-border-left-space | @cell-horizontal-padding | - 
--td-cell-border-right-space | 0 | - 
--td-cell-description-color | @font-gray-2 | - 
--td-cell-description-font-size | @font-size-base | - 
--td-cell-description-line-height | 22px | - 
--td-cell-height | auto | - 
--td-cell-horizontal-padding | 16px | - 
--td-cell-hover-color | @bg-color-secondarycontainer | - 
--td-cell-image-height | 48px | - 
--td-cell-image-width | 48px | - 
--td-cell-left-icon-color | @brand-color | - 
--td-cell-left-icon-font-size | 24px | - 
--td-cell-line-height | 24px | - 
--td-cell-note-color | @font-gray-3 | - 
--td-cell-note-font-size | @font-size-m | - 
--td-cell-required-color | @error-color-6 | - 
--td-cell-required-font-size | @font-size-m | - 
--td-cell-right-icon-color | @font-gray-3 | - 
--td-cell-right-icon-font-size | 24px | - 
--td-cell-title-color | @font-gray-1 | - 
--td-cell-title-font-size | @font-size-m | - 
--td-cell-vertical-padding | 16px | - 
--td-cell-group-border-color | @border-color | - 
--td-cell-group-title-bg-color | @bg-color-secondarycontainer | - 
--td-cell-group-title-color | @font-gray-3 | - 
--td-cell-group-title-font-size | 14px | - 
--td-cell-group-title-line-height | 45px | - 
--td-cell-group-title-padding-left | 16px | - 
