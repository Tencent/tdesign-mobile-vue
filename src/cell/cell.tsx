import { computed, defineComponent } from 'vue';
import { ChevronRightIcon } from 'tdesign-icons-vue-next';
import { Hover } from '../shared';
import config from '../config';
import CellProps from './props';
import { useFormDisabled } from '../form/hooks';
import { usePrefixClass } from '../hooks/useClass';
import { useContent, useTNodeJSX } from '../hooks/tnode';

const { prefix } = config;
const name = `${prefix}-cell`;

export default defineComponent({
  name,
  directives: { Hover },
  props: CellProps,
  emits: ['click'],
  setup(props) {
    const readerTNodeJSX = useTNodeJSX();
    const readerTNodeContent = useContent();
    const disabled = useFormDisabled();
    const cellClass = usePrefixClass('cell');

    const cellClasses = computed(() => [
      `${cellClass.value}`,
      `${cellClass.value}--${props.align}`,
      {
        [`${cellClass.value}--borderless`]: !props.bordered,
      },
    ]);

    const hoverDisabled = computed(() => !props.hover || disabled.value);

    const handleClick = (e: Event) => {
      if (!disabled.value) {
        props.onClick?.(e);
      }
    };

    const readerImage = () => {
      if (typeof props.image === 'string') {
        return <img src={props.image} class={`${cellClass.value}__left-image`} />;
      }
      const image = readerTNodeJSX('image');

      return image;
    };

    const readerLeft = () => {
      const leftIcon = readerTNodeJSX('leftIcon');
      return (
        <div class={`${cellClass.value}__left`}>
          {leftIcon && <div class={`${cellClass.value}__left-icon`}>{leftIcon}</div>}
          {readerImage()}
        </div>
      );
    };
    const readerTitle = () => {
      const title = readerTNodeJSX('title');
      if (!title) {
        return null;
      }
      const description = readerTNodeJSX('description');
      return (
        <div class={`${cellClass.value}__title`}>
          {title}
          {props.required && <span class={`${cellClass.value}--required`}>&nbsp;*</span>}
          {description && <div class={`${cellClass.value}__description`}>{description}</div>}
        </div>
      );
    };
    const readerRight = () => {
      const rightIcon = props.arrow ? <ChevronRightIcon /> : readerTNodeJSX('rightIcon');
      if (!rightIcon) {
        return null;
      }
      return (
        <div class={`${cellClass.value}__right`}>
          <div class={`${cellClass.value}__right-icon`}>{rightIcon}</div>
        </div>
      );
    };

    return () => {
      const note = readerTNodeContent('default', 'note');
      return (
        <div
          v-hover={{ className: `${cellClass.value}--hover`, disabledName: hoverDisabled.value }}
          class={cellClasses.value}
          onClick={handleClick}
        >
          {readerLeft()}
          {readerTitle()}
          {note && <div class={`${cellClass.value}__note`}>{note}</div>}
          {readerRight()}
        </div>
      );
    };
  },
});
