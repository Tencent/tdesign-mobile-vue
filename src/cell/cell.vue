<template>
  <div :class="styleCell" @click="onClick">
    <div :class="`${name}__left`">
      <div v-if="leftIconContent" :class="`${name}__left-icon`">
        <t-node :content="leftIconContent" />
      </div>
      <template v-if="image">
        <img v-if="typeof image === 'string'" :src="image" :class="`${name}__left-image`" />
        <t-node v-else :content="imageContent" />
      </template>
    </div>
    <div v-if="titleContent" :class="`${name}__title`">
      <t-node :content="titleContent" /><span v-if="required" :class="`${name}--required`">&nbsp;*</span>
      <div v-if="descriptionContent" :class="`${name}__description`">
        <t-node :content="descriptionContent" />
      </div>
    </div>
    <div v-if="noteContent" :class="`${name}__note`">
      <t-node :content="noteContent" />
    </div>
    <div v-if="rightIconContent" :class="`${name}__right`">
      <div :class="`${name}__right-icon`">
        <t-node :content="rightIconContent" />
      </div>
    </div>
  </div>
</template>

<script lang="ts">
import { computed, defineComponent, getCurrentInstance, toRefs, h } from 'vue';
import { ChevronRightIcon } from 'tdesign-icons-vue-next';
import { renderTNode, renderContent, TNode, useEmitEvent } from '../shared';
import config from '../config';
import CellProps from './props';

const { prefix } = config;
const name = `${prefix}-cell`;

export default defineComponent({
  name,
  components: { TNode },
  props: CellProps,
  emits: ['click'],
  setup(props, context) {
    const emitEvent = useEmitEvent(props, context.emit);
    const internalInstance = getCurrentInstance();
    const noteContent = computed(() => renderContent(internalInstance, 'default', 'note'));
    const titleContent = computed(() => renderTNode(internalInstance, 'title'));
    const descriptionContent = computed(() => renderTNode(internalInstance, 'description'));

    const chevronRightIcon = h(ChevronRightIcon);
    const rightIconContent = computed(() => {
      if (props.arrow) {
        return chevronRightIcon;
      }
      return renderTNode(internalInstance, 'rightIcon');
    });

    const imageContent = computed(() => renderTNode(internalInstance, 'image'));
    const leftIconContent = computed(() => renderTNode(internalInstance, 'leftIcon', { params: { class: 't' } }));

    const styleCell = computed(() => [
      `${name}`,
      `${name}--${props.align}`,
      {
        [`${name}--hover`]: props.hover,
        [`${name}--bordered`]: props.bordered,
      },
    ]);

    const onClick = (e: Event) => emitEvent('click', e);

    return {
      ...toRefs(props),
      name,
      onClick,
      styleCell,
      imageContent,
      rightIconContent,
      leftIconContent,
      noteContent,
      titleContent,
      descriptionContent,
    };
  },
});
</script>
