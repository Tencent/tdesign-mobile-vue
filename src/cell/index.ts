import _Cell from './cell';
import _CellGroup from './cell-group';
import { withInstall, WithInstallType } from '../shared';
import { TdCellProps, TdCellGroupProps } from './type';

import './style';

export * from './type';
export type CellProps = TdCellProps;
export type CellGroupProps = TdCellGroupProps;

export const Cell: WithInstallType<typeof _Cell> = withInstall(_Cell);
export const CellGroup: WithInstallType<typeof _CellGroup> = withInstall(_CellGroup);
export default Cell;
