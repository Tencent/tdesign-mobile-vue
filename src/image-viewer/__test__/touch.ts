/**
 * https://github.com/vant-ui/vant/blob/dev/packages/vant/test/event.ts
 */

import { ComponentPublicInstance, nextTick } from 'vue';
import { VueWrapper, DOMWrapper } from '@vue/test-utils';

function getTouch(el: Element | Window, x: number, y: number) {
  return {
    identifier: Date.now(),
    target: el,
    pageX: x,
    pageY: y,
    clientX: x,
    clientY: y,
    radiusX: 2.5,
    radiusY: 2.5,
    rotationAngle: 10,
    force: 0.5,
  };
}

// trigger pointer/touch event
export function trigger(
  wrapper: VueWrapper<ComponentPublicInstance<any, any, any>> | DOMWrapper<Element> | Element | Window,
  eventName: string,
  x = 0,
  y = 0,
  options: any = {},
) {
  const el = 'element' in wrapper ? wrapper.element : wrapper;
  const touchList = options.touchList || [getTouch(el, x, y)];

  if (options.x || options.y) {
    touchList.push(getTouch(el, options.x, options.y));
  }

  const event = document.createEvent('CustomEvent');
  event.initCustomEvent(eventName, true, true, {});

  Object.assign(event, {
    clientX: x,
    clientY: y,
    touches: touchList,
    targetTouches: touchList,
    changedTouches: touchList,
  });

  el.dispatchEvent(event);

  return nextTick();
}

function triggerTwoFingerTouchmove(el: Element | DOMWrapper<Element>, x: number, y: number) {
  trigger(el, 'touchmove', -x, -y, { x, y });
}

// simulate zoom
export function triggerZoom(el: Element | DOMWrapper<Element>, x: number, y: number, direction: 'in' | 'out' = 'in') {
  trigger(el, 'touchstart', 0, 0, { x, y });

  if (direction === 'in') {
    triggerTwoFingerTouchmove(el, x / 4, y / 4);
    triggerTwoFingerTouchmove(el, x / 3, y / 3);
    triggerTwoFingerTouchmove(el, x / 2, y / 2);
    triggerTwoFingerTouchmove(el, x, y);
  } else if (direction === 'out') {
    triggerTwoFingerTouchmove(el, x, y);
    triggerTwoFingerTouchmove(el, x / 2, y / 2);
    triggerTwoFingerTouchmove(el, x / 3, y / 3);
    triggerTwoFingerTouchmove(el, x / 4, y / 4);
  }

  trigger(el, 'touchend', 0, 0, { touchList: [] });
}
