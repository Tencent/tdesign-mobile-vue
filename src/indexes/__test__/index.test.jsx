import { mount } from '@vue/test-utils';
import { vi } from 'vitest';
import { default as TCell } from '../../cell/index';
import { Indexes, IndexesAnchor } from '../index';
import config from '@/config';
import { trigger } from '@/image-viewer/__test__/touch';

const { prefix } = config;

const indexesClass = `${prefix}-indexes`;
const indexesAnchorClass = `${prefix}-indexes-anchor`;

const move = async (target, distance) => {
  await trigger(target, 'touchstart', 0, 0);
  await trigger(target, 'touchmove', 0, distance);
  await trigger(target, 'touchend', 0, distance);
};

const children = new Array(5).fill('列表内容');
const list = [
  {
    index: 1,
    children,
  },
  {
    index: 3,
    children,
  },
  {
    index: 5,
    children,
  },
  {
    index: 7,
    children,
  },
  {
    index: 8,
    children,
  },
  {
    index: 10,
    children,
  },
  {
    index: '#',
    children,
  },
];
const indexList = list.map((item) => item.index);

const componentName = `${prefix}-indexes`;
describe('Indexes', () => {
  describe('props', () => {
    it(':indexList', () => {
      const wrapper = mount({
        setup() {
          return () => (
            <Indexes indexList={indexList}>
              {{
                default: list.map((item, index) => (
                  <template>
                    <IndexesAnchor index={item.index} key={index}></IndexesAnchor>
                    {item.children.map((val) => (
                      <TCell className="indexes-cell">{val}</TCell>
                    ))}
                  </template>
                )),
              }}
            </Indexes>
          );
        },
      });
      expect(wrapper.findAll(`.${componentName}__sidebar-item`).length).toBe(indexList.length);
      wrapper.find(`.${componentName}__sidebar-item`).trigger('click');
      expect(wrapper.findAllComponents(TCell).length).toBe(
        list.reduce((count, item) => count + item.children.length, 0),
      );

      const wrapper1 = mount({
        setup() {
          return () => (
            <Indexes>
              {{
                default: list.map((item, index) => (
                  <template>
                    <IndexesAnchor index={item.index} key={index}></IndexesAnchor>
                    {item.children.map((val) => (
                      <TCell className="indexes-cell">{val}</TCell>
                    ))}
                  </template>
                )),
              }}
            </Indexes>
          );
        },
      });
      expect(wrapper1.findAll(`.${componentName}__sidebar-item`).length).toBe(26);
    });
  });
  describe('event', () => {
    it(': select', async () => {
      const selectFn = vi.fn();
      const wrapper = mount({
        components: { Indexes, IndexesAnchor, TCell },
        template: `
          <Indexes :indexList="indexList" @select="select">
            <template v-for="(item, index) in list" :slot="item.index" :key="index">
              <IndexesAnchor :index="item.index"></IndexesAnchor>
              <TCell v-for="val in item.children" class="indexes-cell">{{val}}</TCell>
            </template>
          </Indexes>
        `,
        data() {
          return {
            indexList,
            list,
          };
        },
        methods: {
          select(index) {
            selectFn(index);
          },
        },
      });
      await wrapper.find('.t-indexes__sidebar-item').trigger('click');
      expect(selectFn).toBeCalledWith(list[0].index);
    });

    it(': change', async () => {
      const selectFn = vi.fn();
      const changeFn = vi.fn();
      const wrapper = mount({
        components: { Indexes, IndexesAnchor, TCell },
        template: `
          <Indexes :indexList="indexList" @select="select" @change="change">
            <template v-for="(item, index) in list" :slot="item.index" :key="index">
              <IndexesAnchor :index="item.index"></IndexesAnchor>
              <TCell v-for="val in item.children" class="indexes-cell">{{val}}</TCell>
            </template>
          </Indexes>
        `,
        data() {
          return {
            indexList,
            list,
          };
        },
        methods: {
          select(index) {
            selectFn(index);
          },
          change(index) {
            changeFn(index);
          },
        },
      });
      await wrapper.find(`.${indexesClass}__sidebar-item`).trigger('click');
      expect(selectFn).toBeCalledWith(list[0].index);
      expect(changeFn).toBeCalledWith(list[0].index);

      const items = await wrapper.findAll(`.${indexesClass}__sidebar-item`);
      await items[1].trigger('click');
      expect(selectFn).toBeCalledWith(list[1].index);
      expect(changeFn).toBeCalledWith(list[1].index);
    });

    it(': touch sidebar show tips', async () => {
      const selectFn = vi.fn();
      const wrapper = mount({
        components: { Indexes, IndexesAnchor, TCell },
        template: `
          <Indexes :indexList="indexList" @select="select">
            <template v-for="(item, index) in list" :slot="item.index" :key="index">
              <IndexesAnchor :index="item.index"></IndexesAnchor>
              <TCell v-for="val in item.children" class="indexes-cell">{{val}}</TCell>
            </template>
          </Indexes>
        `,
        data() {
          return {
            indexList,
            list,
          };
        },
        methods: {
          select(index) {
            selectFn(index);
          },
        },
      });
      await wrapper.find('.t-indexes__sidebar-item').trigger('click');
      expect(selectFn).toBeCalledWith(list[0].index);
      expect(wrapper.find('.t-indexes__sidebar-tip')).toBeDefined();
      wrapper.unmount();
    });

    it(': sidebar touchmove', async () => {
      const wrapper = mount({
        components: { Indexes, IndexesAnchor, TCell },
        template: `
          <Indexes :indexList="indexList">
            <template v-for="(item, index) in list" :slot="item.index" :key="index">
              <IndexesAnchor :index="item.index"></IndexesAnchor>
              <TCell v-for="val in item.children" class="indexes-cell">{{val}}</TCell>
            </template>
          </Indexes>
        `,
        data() {
          return {
            indexList,
            list,
          };
        },
      });
      const sidebar = wrapper.find(`.${componentName}__sidebar`);
      document.elementFromPoint = function () {
        return sidebar.findAll(`.${componentName}__sidebar-item`)[0].element;
      };
      await trigger(sidebar, 'touchstart', 20, 20);
      await trigger(sidebar, 'touchmove', 20, 60);
      await trigger(sidebar, 'touchend', 20, 310);
    });
  });
});

describe('Indexes & IndexesAnchor', () => {
  it('render', () => {
    const wrapper = mount({
      setup() {
        return () => (
          <Indexes indexList={indexList}>
            {{
              default: list.map((item, index) => (
                <template>
                  <IndexesAnchor index={item.index} key={index}></IndexesAnchor>
                  {item.children.map((val) => (
                    <TCell className="indexes-cell">{val}</TCell>
                  ))}
                </template>
              )),
            }}
          </Indexes>
        );
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});

describe('event', () => {
  it('@touch', async () => {
    const onSelect = vi.fn();
    const onChange = vi.fn();
    const wrapper = mount(Indexes, {
      props: {
        indexList,
        onSelect,
        onChange,
      },
    });

    const sidebar = wrapper.find(`.${componentName}__sidebar`);

    const $sideBarItems = wrapper.findAll(`.${componentName}__sidebar-item`);
    // 模拟点击第一项 item, 此时第一项为激活态，两个事件均被触发
    await $sideBarItems[0].trigger('click');
    expect(onChange).toBeCalledTimes(1);
    expect(onSelect).toBeCalledTimes(1);
    expect($sideBarItems[0].classes()).toContain(`${componentName}__sidebar-item--active`);

    // mock document.elementFromPoint
    document.elementFromPoint = function () {
      return sidebar.findAll(`.${componentName}__sidebar-item`)[2].element;
    };

    // 每一项的高度是20，这里以第0项为起始点，模拟滑动 40，此时第3项转为激活态，只会触发 change 事件
    await move($sideBarItems[0], 40);
    expect(wrapper.findAll(`.${componentName}__sidebar-item`)[2].classes()).toContain(
      `${componentName}__sidebar-item--active`,
    );
    expect(onSelect).toBeCalledTimes(1);
    expect(onChange).toBeCalledTimes(2);
  });
});
