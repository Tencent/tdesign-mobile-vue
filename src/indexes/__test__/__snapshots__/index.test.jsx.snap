// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Indexes & IndexesAnchor > render 1`] = `
<div
  class="t-indexes"
>
  <div
    class="t-indexes__sidebar"
  >
    
    <div
      class="t-indexes__sidebar-item"
      data-index="1"
    >
      1
      <!---->
    </div>
    <div
      class="t-indexes__sidebar-item"
      data-index="3"
    >
      3
      <!---->
    </div>
    <div
      class="t-indexes__sidebar-item"
      data-index="5"
    >
      5
      <!---->
    </div>
    <div
      class="t-indexes__sidebar-item"
      data-index="7"
    >
      7
      <!---->
    </div>
    <div
      class="t-indexes__sidebar-item"
      data-index="8"
    >
      8
      <!---->
    </div>
    <div
      class="t-indexes__sidebar-item"
      data-index="10"
    >
      10
      <!---->
    </div>
    <div
      class="t-indexes__sidebar-item"
      data-index="#"
    >
      #
      <!---->
    </div>
    
  </div>
  
  <template>
    <div
      class="t-indexes-anchor"
      data-index="1"
    >
      <div
        class="t-indexes-anchor__wrapper"
      >
        <div
          class="t-indexes-anchor__slot"
        >
          <!---->
        </div>
        <div
          class="t-indexes-anchor__header"
        >
          1
        </div>
      </div>
    </div>
    
    <div
      class="indexes-cell"
    >
      <div
        class="t-cell__left"
      >
        <!---->
        <!---->
      </div>
      <!---->
      <div
        class="t-cell__note"
      >
        
        列表内容
        
      </div>
      <!---->
    </div>
    <div
      class="indexes-cell"
    >
      <div
        class="t-cell__left"
      >
        <!---->
        <!---->
      </div>
      <!---->
      <div
        class="t-cell__note"
      >
        
        列表内容
        
      </div>
      <!---->
    </div>
    <div
      class="indexes-cell"
    >
      <div
        class="t-cell__left"
      >
        <!---->
        <!---->
      </div>
      <!---->
      <div
        class="t-cell__note"
      >
        
        列表内容
        
      </div>
      <!---->
    </div>
    <div
      class="indexes-cell"
    >
      <div
        class="t-cell__left"
      >
        <!---->
        <!---->
      </div>
      <!---->
      <div
        class="t-cell__note"
      >
        
        列表内容
        
      </div>
      <!---->
    </div>
    <div
      class="indexes-cell"
    >
      <div
        class="t-cell__left"
      >
        <!---->
        <!---->
      </div>
      <!---->
      <div
        class="t-cell__note"
      >
        
        列表内容
        
      </div>
      <!---->
    </div>
    
  </template>
  <template>
    <div
      class="t-indexes-anchor"
      data-index="3"
    >
      <div
        class="t-indexes-anchor__wrapper"
      >
        <div
          class="t-indexes-anchor__slot"
        >
          <!---->
        </div>
        <div
          class="t-indexes-anchor__header"
        >
          3
        </div>
      </div>
    </div>
    
    <div
      class="indexes-cell"
    >
      <div
        class="t-cell__left"
      >
        <!---->
        <!---->
      </div>
      <!---->
      <div
        class="t-cell__note"
      >
        
        列表内容
        
      </div>
      <!---->
    </div>
    <div
      class="indexes-cell"
    >
      <div
        class="t-cell__left"
      >
        <!---->
        <!---->
      </div>
      <!---->
      <div
        class="t-cell__note"
      >
        
        列表内容
        
      </div>
      <!---->
    </div>
    <div
      class="indexes-cell"
    >
      <div
        class="t-cell__left"
      >
        <!---->
        <!---->
      </div>
      <!---->
      <div
        class="t-cell__note"
      >
        
        列表内容
        
      </div>
      <!---->
    </div>
    <div
      class="indexes-cell"
    >
      <div
        class="t-cell__left"
      >
        <!---->
        <!---->
      </div>
      <!---->
      <div
        class="t-cell__note"
      >
        
        列表内容
        
      </div>
      <!---->
    </div>
    <div
      class="indexes-cell"
    >
      <div
        class="t-cell__left"
      >
        <!---->
        <!---->
      </div>
      <!---->
      <div
        class="t-cell__note"
      >
        
        列表内容
        
      </div>
      <!---->
    </div>
    
  </template>
  <template>
    <div
      class="t-indexes-anchor"
      data-index="5"
    >
      <div
        class="t-indexes-anchor__wrapper"
      >
        <div
          class="t-indexes-anchor__slot"
        >
          <!---->
        </div>
        <div
          class="t-indexes-anchor__header"
        >
          5
        </div>
      </div>
    </div>
    
    <div
      class="indexes-cell"
    >
      <div
        class="t-cell__left"
      >
        <!---->
        <!---->
      </div>
      <!---->
      <div
        class="t-cell__note"
      >
        
        列表内容
        
      </div>
      <!---->
    </div>
    <div
      class="indexes-cell"
    >
      <div
        class="t-cell__left"
      >
        <!---->
        <!---->
      </div>
      <!---->
      <div
        class="t-cell__note"
      >
        
        列表内容
        
      </div>
      <!---->
    </div>
    <div
      class="indexes-cell"
    >
      <div
        class="t-cell__left"
      >
        <!---->
        <!---->
      </div>
      <!---->
      <div
        class="t-cell__note"
      >
        
        列表内容
        
      </div>
      <!---->
    </div>
    <div
      class="indexes-cell"
    >
      <div
        class="t-cell__left"
      >
        <!---->
        <!---->
      </div>
      <!---->
      <div
        class="t-cell__note"
      >
        
        列表内容
        
      </div>
      <!---->
    </div>
    <div
      class="indexes-cell"
    >
      <div
        class="t-cell__left"
      >
        <!---->
        <!---->
      </div>
      <!---->
      <div
        class="t-cell__note"
      >
        
        列表内容
        
      </div>
      <!---->
    </div>
    
  </template>
  <template>
    <div
      class="t-indexes-anchor"
      data-index="7"
    >
      <div
        class="t-indexes-anchor__wrapper"
      >
        <div
          class="t-indexes-anchor__slot"
        >
          <!---->
        </div>
        <div
          class="t-indexes-anchor__header"
        >
          7
        </div>
      </div>
    </div>
    
    <div
      class="indexes-cell"
    >
      <div
        class="t-cell__left"
      >
        <!---->
        <!---->
      </div>
      <!---->
      <div
        class="t-cell__note"
      >
        
        列表内容
        
      </div>
      <!---->
    </div>
    <div
      class="indexes-cell"
    >
      <div
        class="t-cell__left"
      >
        <!---->
        <!---->
      </div>
      <!---->
      <div
        class="t-cell__note"
      >
        
        列表内容
        
      </div>
      <!---->
    </div>
    <div
      class="indexes-cell"
    >
      <div
        class="t-cell__left"
      >
        <!---->
        <!---->
      </div>
      <!---->
      <div
        class="t-cell__note"
      >
        
        列表内容
        
      </div>
      <!---->
    </div>
    <div
      class="indexes-cell"
    >
      <div
        class="t-cell__left"
      >
        <!---->
        <!---->
      </div>
      <!---->
      <div
        class="t-cell__note"
      >
        
        列表内容
        
      </div>
      <!---->
    </div>
    <div
      class="indexes-cell"
    >
      <div
        class="t-cell__left"
      >
        <!---->
        <!---->
      </div>
      <!---->
      <div
        class="t-cell__note"
      >
        
        列表内容
        
      </div>
      <!---->
    </div>
    
  </template>
  <template>
    <div
      class="t-indexes-anchor"
      data-index="8"
    >
      <div
        class="t-indexes-anchor__wrapper"
      >
        <div
          class="t-indexes-anchor__slot"
        >
          <!---->
        </div>
        <div
          class="t-indexes-anchor__header"
        >
          8
        </div>
      </div>
    </div>
    
    <div
      class="indexes-cell"
    >
      <div
        class="t-cell__left"
      >
        <!---->
        <!---->
      </div>
      <!---->
      <div
        class="t-cell__note"
      >
        
        列表内容
        
      </div>
      <!---->
    </div>
    <div
      class="indexes-cell"
    >
      <div
        class="t-cell__left"
      >
        <!---->
        <!---->
      </div>
      <!---->
      <div
        class="t-cell__note"
      >
        
        列表内容
        
      </div>
      <!---->
    </div>
    <div
      class="indexes-cell"
    >
      <div
        class="t-cell__left"
      >
        <!---->
        <!---->
      </div>
      <!---->
      <div
        class="t-cell__note"
      >
        
        列表内容
        
      </div>
      <!---->
    </div>
    <div
      class="indexes-cell"
    >
      <div
        class="t-cell__left"
      >
        <!---->
        <!---->
      </div>
      <!---->
      <div
        class="t-cell__note"
      >
        
        列表内容
        
      </div>
      <!---->
    </div>
    <div
      class="indexes-cell"
    >
      <div
        class="t-cell__left"
      >
        <!---->
        <!---->
      </div>
      <!---->
      <div
        class="t-cell__note"
      >
        
        列表内容
        
      </div>
      <!---->
    </div>
    
  </template>
  <template>
    <div
      class="t-indexes-anchor"
      data-index="10"
    >
      <div
        class="t-indexes-anchor__wrapper"
      >
        <div
          class="t-indexes-anchor__slot"
        >
          <!---->
        </div>
        <div
          class="t-indexes-anchor__header"
        >
          10
        </div>
      </div>
    </div>
    
    <div
      class="indexes-cell"
    >
      <div
        class="t-cell__left"
      >
        <!---->
        <!---->
      </div>
      <!---->
      <div
        class="t-cell__note"
      >
        
        列表内容
        
      </div>
      <!---->
    </div>
    <div
      class="indexes-cell"
    >
      <div
        class="t-cell__left"
      >
        <!---->
        <!---->
      </div>
      <!---->
      <div
        class="t-cell__note"
      >
        
        列表内容
        
      </div>
      <!---->
    </div>
    <div
      class="indexes-cell"
    >
      <div
        class="t-cell__left"
      >
        <!---->
        <!---->
      </div>
      <!---->
      <div
        class="t-cell__note"
      >
        
        列表内容
        
      </div>
      <!---->
    </div>
    <div
      class="indexes-cell"
    >
      <div
        class="t-cell__left"
      >
        <!---->
        <!---->
      </div>
      <!---->
      <div
        class="t-cell__note"
      >
        
        列表内容
        
      </div>
      <!---->
    </div>
    <div
      class="indexes-cell"
    >
      <div
        class="t-cell__left"
      >
        <!---->
        <!---->
      </div>
      <!---->
      <div
        class="t-cell__note"
      >
        
        列表内容
        
      </div>
      <!---->
    </div>
    
  </template>
  <template>
    <div
      class="t-indexes-anchor"
      data-index="#"
    >
      <div
        class="t-indexes-anchor__wrapper"
      >
        <div
          class="t-indexes-anchor__slot"
        >
          <!---->
        </div>
        <div
          class="t-indexes-anchor__header"
        >
          #
        </div>
      </div>
    </div>
    
    <div
      class="indexes-cell"
    >
      <div
        class="t-cell__left"
      >
        <!---->
        <!---->
      </div>
      <!---->
      <div
        class="t-cell__note"
      >
        
        列表内容
        
      </div>
      <!---->
    </div>
    <div
      class="indexes-cell"
    >
      <div
        class="t-cell__left"
      >
        <!---->
        <!---->
      </div>
      <!---->
      <div
        class="t-cell__note"
      >
        
        列表内容
        
      </div>
      <!---->
    </div>
    <div
      class="indexes-cell"
    >
      <div
        class="t-cell__left"
      >
        <!---->
        <!---->
      </div>
      <!---->
      <div
        class="t-cell__note"
      >
        
        列表内容
        
      </div>
      <!---->
    </div>
    <div
      class="indexes-cell"
    >
      <div
        class="t-cell__left"
      >
        <!---->
        <!---->
      </div>
      <!---->
      <div
        class="t-cell__note"
      >
        
        列表内容
        
      </div>
      <!---->
    </div>
    <div
      class="indexes-cell"
    >
      <div
        class="t-cell__left"
      >
        <!---->
        <!---->
      </div>
      <!---->
      <div
        class="t-cell__note"
      >
        
        列表内容
        
      </div>
      <!---->
    </div>
    
  </template>
  
</div>
`;
